src/components/useCurrencyConverter.js

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from "react";
+
+export default function useCurrencyConverter(endpoint, access_key) {
+  const [currencyOptions, setCurrencyOptions] = useState([]);
+  const [fromCurrency, setFromCurrency] = useState();
+  const [toCurrency, setToCurrency] = useState();
+  const [exchangeRate, setExchangeRate] = useState();
+  const [amount, setAmount] = useState(1);
+  const [isAmountInFromCurrency, setAmountInFromCurrency] = useState(true);
+  const [currencies, setCurrencies] = useState({});
+  const [timestamp, setTimestamp] = useState();
+
+  useEffect(() => {
+    fetch(
+      `http://api.exchangeratesapi.io/v1/${endpoint}?access_key=${access_key}`
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        setCurrencies(data);
+        setTimestamp(data.timestamp);
+        const initialCurrency = Object.keys(data.rates)[0];
+        setCurrencyOptions([data.base, ...Object.keys(data.rates)]);
+        setFromCurrency(data.base);
+        setToCurrency(initialCurrency);
+        setExchangeRate(data.rates[initialCurrency]);
+      });
+  }, []);
```

Comment:
How does this handle caching? It could be beneficial to take a look at react useContext.